File: 05-Full-stack-event-management/int-poc-eventmgt-web-dev/app/(dashboard)/events/[id]/edit/steps/schemas/venue.schema.js
import * as yup from "yup";

export const venueSchema = yup.object({
    venue: yup.object({
        name: yup.string().required(),
        address: yup.string().required(),
        city: yup.string().required(),
        country: yup.string().required(),

        state: yup.string().nullable(),
        map_link: yup.string().url().nullable(),
        landmark: yup.string().nullable(),

        has_parking: yup.boolean().required(),
        gates_count: yup
            .number()
            .required()
            .min(1),

        seat_plan_image: yup
            .mixed()
            .required("Seat plan image is required"),

        map_image: yup
            .mixed()
            .required("Venue map image is required"),

        by_car: yup.string().nullable(),
        by_metro: yup.string().nullable(),
        by_bus: yup.string().nullable(),
    }),
});

File: 05-Full-stack-event-management/int-poc-eventmgt-api-dev/app/schemas/venue_schema.py
from typing import Optional
from uuid import UUID
from pydantic import BaseModel


class VenueUpsert(BaseModel):
    event_id: UUID

    name: str
    address: str
    city: str
    state: Optional[str] = None
    country: str

    map_link: Optional[str] = None
    landmark: Optional[str] = None

    has_parking: bool
    gates_count: int

    seat_plan_image_url: str
    map_image_url: str

    by_car: Optional[str] = None
    by_metro: Optional[str] = None
    by_bus: Optional[str] = None
